core/src/plugins/filed/python/vmware/BareosFdPluginVMware.py

```diff
@@ -1116,6 +1268,326 @@ def _get_dcftree(self, dcf, folder, vm_folder):
+        vm_folder_view.Destroy()
+
+    def find_or_create_vm_folder(self, folder_name):
+        all_vm_folders = {}
+        self._get_vm_folders(all_vm_folders, "/", self.dc.vmFolder)
+        if folder_name not in all_vm_folders:
+            bareosfd.JobMessage(
+                bareosfd.M_INFO,
+                "VM folder %s not found, creating it\n"
+                % (StringCodec.encode(folder_name)),
+            )
+            # the prefixed "/" will produce an empty string element in split result,
+            # so we must skip element 0
+            current_folder_path = "/"
+            current_folder = self.dc.vmFolder
+            for folder_part in folder_name.split("/")[1:]:
+                current_folder_path += "/" + folder_part
+                # avoid leading "//"
+                current_folder_path = "/" + current_folder_path.lstrip("/")
+                if current_folder_path in all_vm_folders:
+                    current_folder = all_vm_folders[current_folder_path]
+                    continue
+                current_folder = current_folder.CreateFolder(folder_part)
+                all_vm_folders[current_folder_path] = current_folder
+
+        return all_vm_folders[folder_name]
+
+    def get_objects_by_folder(self, start_folder, folder_name, objects_by_folder):
+        """
+        Generic Recursive function to get objects by folder.
+        Here, folder means a folder path, using "/" as a folder separator.
+        An empty dict must be passed initially, it will be filled with
+        folder path as key and managed object as value.
+        """
+        for current_object in start_folder.childEntity:
+            if isinstance(current_object, vim.Folder):
+                self.get_objects_by_folder(
+                    current_object,
+                    folder_name + "/" + current_object.name,
+                    objects_by_folder,
+                )
+            else:
+                objects_by_folder[
+                    "/" + (folder_name + "/").lstrip("/") + current_object.name
+                ] = current_object
+
+        return
+
+    def _get_resource_pools(self, cluster):
+        """
+        Get resource pools by folder like structure
+
+        Resource pools can be nested. Each cluster has a default resource pool
+        which is its resourcePool property and it's a single item.
+        But the resourcePool property of the default resource pool is a list
+        of resource pools, where each of this can have a list of resource pools.
+        Like this, resource pools can be arbitrarily nested.
+        """
+        resource_pools = {}
+        resource_pools["/"] = cluster.resourcePool
+        stack = []
+        for resource_pool in cluster.resourcePool.resourcePool:
+            stack.append(("/", resource_pool))
+        while stack:
+            folder, resource_pool = stack.pop()
+            path = "/" + (folder + "/").lstrip("/") + resource_pool.name
```

Comment:
This trick to avoid double slashes is done multiple times. Is this the best way to do it, and does it make sense to write a function for it?